### PR TITLE
matter: merge upstream commits up to 9012f0

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Do a regular upmerge and align matter samples accordingly.

**NOTE**
Like always, files in `zap-generated` (formerly `gen`) directory are auto-generated and don't need to be reviewed.

### CODE REVIEW

File: applications/matter_weather_station/src/app_task.cpp [hunk 1]
```diff
@@ -120,19 +125,18 @@ void AppTask::OpenPairingWindow()
...
 	}
 
 	if (ConnectivityMgr().IsBLEAdvertisingEnabled()) {
-		LOG_INF("BLE Advertisement is already enabled");
+		LOG_INF("BLE advertising is already enabled");
 		return;
 	}
 
-	if (OpenDefaultPairingWindow(chip::ResetFabrics::kNo) == CHIP_NO_ERROR) {
-		LOG_INF("Enabled BLE Advertisement");
```
Comment (kkasperczyk-no) [inline applications/matter_weather_station/src/app_task.cpp:133]: Don't you think that log is useful? I mean it seems to be a good user experience if you want to do something by pressing a button and get a log confirming it went fine.

**Review by kkasperczyk-no — APPROVED** [review]
LGTM, just one comment to consider :+1:

File: applications/matter_weather_station/src/app_task.cpp (hunk 1 above)
Comment (Damian-Nordic) [reply applications/matter_weather_station/src/app_task.cpp:133]: The reason I removed this log is that the device layer adds two more log messages about starting BLE advertising, so thought another one was useless :)

Comment (kkasperczyk-no) [reply applications/matter_weather_station/src/app_task.cpp:133]: Oh ok, nevermind then

**Review by LuDuda — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -118,11 +118,14 @@ The current |NCS| master branch is based on the Zephyr v2.7 development branch.
...
-The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from the upstream Matter repository up to, and including, ``b77bfb047374b7013dbdf688f542b9326842a39e``.
+The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from the upstream Matter repository up to, and including, ``9012f08de9b7340e7d59d51a7ec8a6cdcfda9d15``.
 
 The following list summarizes the most important changes inherited from the upstream Matter:
 
-* No changes yet
+* Added:
+
+  * Support for Administrator Commissioning Cluster which allows to enable or disable the commissioning window on a Matter device.
+    It is required by the Matter multi-admin functionality.
```
Comment (wiba-nordic) [inline doc/nrf/releases/release-notes-changelog.rst:128]:
```suggestion
  * Support for Administrator Commissioning Cluster, which allows enabling or disabling the commissioning window on a Matter device.
    This is required by the Matter multi-admin functionality.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
I little change needed.

**Review by wiba-nordic — APPROVED** [review]
Looks good

**Pull request merged**